### PR TITLE
solution

### PR DESCRIPTION
**Pull request by serentinos**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -20,6 +20,21 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
```
Comment (33Orange) [inline src/averageAges.js:22]: Clean up the code from unnecessary comments.

File: src/averageAges.js [hunk 2]
```diff
@@ -20,6 +20,21 @@ function calculateMenAverageAge(people, century) {
...
+      const personCenturyDeath = Math.ceil(person.died / 100);
+      const personSex = person.sex;
+
+      return century
+        ? personSex === 'm' && century === personCenturyDeath
+        : personSex === 'm';
+    });
+
+  const averageMenAge = mens
+    .reduce((prev, { born, died }) => prev + (died - born), 0) / mens.length
```
Comment (33Orange) [inline src/averageAges.js:34]: You do the same calculations with reduce in every function. Maybe we should move this calculation to a separate helper?

File: src/averageAges.js [hunk 3]
```diff
@@ -31,13 +46,25 @@ function calculateMenAverageAge(people, century) {
...
+    ? people
+      .filter(person => (
+        people.find(child => child.mother === person.name && person.sex === 'f')
+      ))
+    : people
+      .filter(person => person.sex === 'f');
+
+  const averageAge = filteredWoman
+    .reduce((sum, { born, died }) => sum + (died - born), 0)
+      / filteredWoman.length;
```
Comment (33Orange) [inline src/averageAges.js:65]: You do the same calculations with reduce in every function. Maybe we should move this calculation to a separate helper?

File: src/averageAges.js [hunk 4]
```diff
@@ -49,13 +76,33 @@ function calculateWomenAverageAge(people, withChildren) {
...
+    const ageDifference = personChildren
+      .map(child => child.born - person.born);
+
+    differences
+      .push(...ageDifference);
+  });
+
+  const avgDifference = differences
+    .reduce((sum, current) => sum + current, 0)
+    / differences.length;
```
Comment (33Orange) [inline src/averageAges.js:103]: You do the same calculations with reduce in every function. Maybe we should move this calculation to a separate helper?

File: src/averageAges.js [hunk 5]
```diff
@@ -31,13 +46,25 @@ function calculateMenAverageAge(people, century) {
...
+//  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWoman = withChildren
+    ? people
+      .filter(person => (
+        people.find(child => child.mother === person.name && person.sex === 'f')
+      ))
+    : people
+      .filter(person => person.sex === 'f');
```
Comment (33Orange) [inline src/averageAges.js:61]: You use a filter in both cases. Maybe it's better to do a check `withChildren` condition inside the filter?

File: src/averageAges.js [hunk 6]
```diff
@@ -49,13 +76,33 @@ function calculateWomenAverageAge(people, withChildren) {
...
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const differences = [];
+
+  people.forEach(person => {
+    const personChildren = onlyWithSon
+      ? people
+        .filter(child => person.name === child.mother && child.sex === 'm')
+      : people
+        .filter(child => person.name === child.mother);
```
Comment (33Orange) [inline src/averageAges.js:92]: You use a filter in both cases. Maybe it's better to do a check `onlyWithSon` condition inside the filter?

**Review by 33Orange — CHANGES_REQUESTED** [review]
Well done!

A few moment with readability

**Comment by serentinos** [comment]
@33Orange Thank you for all your suggestions! This advice is really helpful!

**Review by 33Orange — APPROVED** [review]
Well done ! 🔥

**Pull request closed without merging**